arbnode/inbox_tracker.go

```diff
@@ -253,31 +254,47 @@ func (t *InboxTracker) legacyGetDelayedMessageAndAccumulator(seqNum uint64) (*ar
 	return msg, acc, err
 }
 
-func (t *InboxTracker) GetDelayedMessageAndAccumulator(seqNum uint64) (*arbostypes.L1IncomingMessage, common.Hash, error) {
-	key := dbKey(rlpDelayedMessagePrefix, seqNum)
-	exists, err := t.db.Has(key)
+func (t *InboxTracker) GetDelayedMessageAccumulatorAndParentChainBlockNumber(seqNum uint64) (*arbostypes.L1IncomingMessage, common.Hash, uint64, error) {
+	delayedMessageKey := dbKey(rlpDelayedMessagePrefix, seqNum)
+	exists, err := t.db.Has(delayedMessageKey)
 	if err != nil {
-		return nil, common.Hash{}, err
+		return nil, common.Hash{}, 0, err
 	}
 	if !exists {
-		return t.legacyGetDelayedMessageAndAccumulator(seqNum)
+		msg, acc, err := t.legacyGetDelayedMessageAndAccumulator(seqNum)
+		return msg, acc, 0, err
 	}
-	data, err := t.db.Get(key)
+	data, err := t.db.Get(delayedMessageKey)
 	if err != nil {
-		return nil, common.Hash{}, err
+		return nil, common.Hash{}, 0, err
 	}
 	if len(data) < 32 {
-		return nil, common.Hash{}, errors.New("delayed message new entry missing accumulator")
+		return nil, common.Hash{}, 0, errors.New("delayed message new entry missing accumulator")
 	}
 	var acc common.Hash
 	copy(acc[:], data[:32])
 	var msg *arbostypes.L1IncomingMessage
 	err = rlp.DecodeBytes(data[32:], &msg)
-	return msg, acc, err
+	if err != nil {
+		return msg, acc, 0, err
+	}
+
+	parentChainBlockNumberKey := dbKey(parentChainBlockNumberPrefix, seqNum)
+	exists, err = t.db.Has(parentChainBlockNumberKey)
+	if err != nil || !exists {
+		return msg, acc, 0, nil //nolint:nilerr
+	}
+	data, err = t.db.Get(parentChainBlockNumberKey)
+	if err != nil {
+		return msg, acc, 0, nil //nolint:nilerr
```

Comment:
If we encounter an error in either of these places, we should return it. We should return `msg, acc, msg.Header.BlockNumber, nil` if `!exists && error == nil`

Reply:
done.